media/js/newsletter/management.es6.js

```diff
@@ -0,0 +1,784 @@
+            if (keyField === 'order') {
+                return a.order - b.order;
+            } else {
+                const titleA = a.title.toLowerCase();
+                const titleB = b.title.toLowerCase();
+
+                if (titleA < titleB) {
+                    return -1;
+                }
+                if (titleA > titleB) {
+                    return 1;
+                }
+                return 0;
+            }
+        });
+    },
+
+    /**
+     * Fetch JSON object of translated newsletter strings from Bedrock.
+     * @returns {Promise}
+     */
+    getNewsletterStrings: () => {
+        return new window.Promise((resolve, reject) => {
+            const url = NewsletterManagementForm.getNewsletterStringsURL();
+
+            _fetch(url, 'GET')
+                .then((resp) => {
+                    resolve(resp);
+                })
+                .catch((e) => {
+                    reject(e);
+                });
+        });
+    },
+
+    /**
+     * Fetch JSON object of all available newsletters from Basket.
+     * @returns {Promise}
+     */
+    getNewsletterData: () => {
+        return new window.Promise((resolve, reject) => {
+            const url = NewsletterManagementForm.getNewsletterDataURL();
+
+            _fetch(url, 'GET')
+                .then((resp) => {
+                    resolve(resp.newsletters);
+                })
+                .catch((e) => {
+                    reject(e);
+                });
+        });
+    },
+
+    /**
+     * Fetch JSON object of user data from Basket
+     * @returns {Promise}
+     */
+    getUserData: () => {
+        return new window.Promise((resolve, reject) => {
+            const hasFxA =
+                NewsletterManagementForm.getPageURL().indexOf('fxa') !== -1;
+            const action = NewsletterManagementForm.getUserDataURL();
+            const url = hasFxA ? action : action + '?fxa=1';
+
+            _fetch(url, 'GET')
+                .then((resp) => {
+                    // if `has_fxa` not returned from basket, set it from the URL
+                    if (!resp.has_fxa) {
+                        resp.has_fxa = hasFxA;
+                    }
+                    resolve(resp);
+                })
+                .catch((e) => {
+                    reject(e);
+                });
+        });
+    },
+
+    /**
+     * Generate HTML markup for newsletter table row.
+     * @param {Object} newsletter
+     * @param {Number} index
+     * @returns {String}
+     */
+    renderTableRow: (newsletter, index) => {
+        const checked = newsletter.subscribed ? ' checked=""' : '';
+        const indent = newsletter.indent ? ' class="indented"' : '';
+        return `<tr${indent}>
+            <th>
+                <h4>${newsletter.title}</h4>
+                <p>${newsletter.description}</p>
+            </th>
+            <td>
+                <label for="id_form-${index}-subscribed_check">${newsletter.subscribeCopy}</label>
```

Comment:
**issue (a11y, pre-existing)**: we don't get a descriptive label for screenreaders to indicate what newsletter is being subscribed to
<img width="531" alt="unhelpful-label" src="https://user-images.githubusercontent.com/19650432/206213864-9455301a-24f6-4a2d-a427-0f4f41d4b021.png">
<img width="514" alt="vo-testing" src="https://user-images.githubusercontent.com/19650432/206213894-e99e2feb-8427-4954-8801-38a211075054.png">

**suggestion**: probably best spun out into a separate issue, maybe simplest fix is to add a visually hidden span after the subscribeCopy with title

Reply:
Hmm yeah, I think this one warrants a separate issue. Each newsletter is described with an accessible heading and description, however they are in a separate table cell. We might be able to improve how the markup is structured here.